### PR TITLE
feat(cubejs-client-core): Fill missing dates with custom measure value

### PR DESCRIPTION
**Pull request by nagulan23**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

**Issue Reference this PR resolves**

https://github.com/cube-js/cube/issues/4604
https://github.com/cube-js/cube/issues/718

**Description of Changes Made**

Opens flexibility for users to provide custom value to be filled for the missing date.

### CODE REVIEW

**Comment by igorlukanin** [comment]
Thanks for your contribution, @nagulan23 🙌 

Could you also please update the docs on this page? https://cube.dev/docs/reference/frontend/cubejs-client-core

**Comment by nagulan23** [comment]
@igorlukanin Have updated the doc.

File: packages/cubejs-client-core/index.d.ts [hunk 1]
```diff
@@ -252,9 +252,13 @@ declare module '@cubejs-client/core' {
...
     y?: string[];
     /**
-     * If `true` missing dates on the time dimensions will be filled with `0` for all measures.Note: the `fillMissingDates` option set to `true` will override any **order** applied to the query
+     * If `true` missing dates on the time dimensions will be filled with fillWithValue or `0` by default for all measures.Note: the `fillMissingDates` option set to `true` will override any **order** applied to the query
      */
     fillMissingDates?: boolean | null;
+    /**
+     * Value to autofill all the missing date's measure.
+     */
+    fillWithValue?: string | null;
```
Comment (KSDaemon) [inline packages/cubejs-client-core/index.d.ts:261]: Maybe it should be Number instead?

Comment (KSDaemon) [reply packages/cubejs-client-core/index.d.ts:261]: And even in tests you use `5` as number, and not as string.

**Comment by KSDaemon** [comment]
Jentle ping @nagulan23

File: packages/cubejs-client-core/index.d.ts (hunk 1 above)
Comment (nagulan23) [reply packages/cubejs-client-core/index.d.ts:261]: @KSDaemon It was meant to be a string. For example lets take an example of users need the auto filled values to be '-' or 'N/A' or 'Missing'. So this enables a scope of allowing users to fill in the missing measure with any value: Number / String.

Comment (nagulan23) [reply packages/cubejs-client-core/index.d.ts:261]: adding an extra testcase of string example

Comment (KSDaemon) [inline packages/cubejs-client-core/index.d.ts:261]: Thnx for adding another test! 
As number is still a valid type for this, let's updat the typedef too:
```suggestion
    fillWithValue?: string | number | null;
```

**Pull request merged**